docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+Furthermore, it assumes that:
+
+- Community owner nodes have enough storage to persist historical messages older than 30 days
+- Community owner nodes provide archives with historical messages **at least** every 30 days
+- Community owner nodes receive all community messages
+- Community owner nodes are honest
+
+## Overview
+
+The following is a high-level overview of the user flow and features this specification describes.
+
+### Serving community history archives
+
+Community owner nodes go through the following (high level) process to provide community members with message histories (assumes community owner node is available 24/7):
+
+1. Community owner creates a Status community
+2. Community owner enables community history archive support
+3. A special type of channel for distributing magnet links ([Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme), [Extensions for Peers to Send Metadata Files](https://www.bittorrent.org/beps/bep_0009.html)) is created
+4. Community owner invites members and creates additional channels
+5. Community owner node receives messages and stores them into local database
+6. After 7 days, the community owner node exports and compresses last 7 days worth of messages from database and creates a magnet link from that data via torrent client
+7. Community owner node creates message archive index and bundles the previously generated magnet link into it
+8. Community owner node creates magnet link from index and distributes it to community members via special channel created in step 2) through the Waku network
+9. Every subsequent 7 days, the step 6) - 8) are repeated and bundled with the previously distributed archive
+
+### Serving archives for missed messages
+
+If the community owner node goes offline, it MUST go through the following process:
+
+1. Community owner node restarts
+2. Community owner node requests messages from store nodes for the missed time range
+3. Missed messages are stored into local database
+4. Community owner node creates message archive and magnet link for missed messages
+5. Community oowner node creates new message archive index bundled with newly created magnet link
+6. The new index is is distributed to community members via special channel through the Waku network
+
+### Receiving community history archives
+
+Community member nodes go through the following (high level) process to fetch and restore community message histories:
+
+1. User joins community and becomes community member
+2. By joining a community, member nodes automatically subscribe to special magnet link channel provided by the community
+3. Member node requests message history (last 30 days) of community channels from store nodes
+4. Member node receives magnet link message from store nodes
+5. Member node extracts magnet link from message and passes it to torrent client
+6. Torrent client downloads latest message archive index via magnet link
+7. Member node fetches missing archives via torrent
```

Comment:
Perhaps change from 

"7. Member node fetches missing archives via torrent"

to 

""7. Member node fetches binary blob that contains the message history archive for the community via torrent"